src/hooks/useLastAccessedReportID.ts

```diff
@@ -123,12 +123,15 @@ function addSubscriber(subscriber: () => void) {
 
 /**
  * Get the last accessed reportID.
+ * @param skip - Whether to skip the calculation of last accessed report ID
  */
-export default function useLastAccessedReportID(shouldOpenOnAdminRoom: boolean) {
+export default function useLastAccessedReportID(shouldOpenOnAdminRoom: boolean, skip: boolean) {
```

Comment:
Rather than adding a `skip` boolean param in a hook, let's just call `ReportUtils.findLastAccessedReport` in a `useEffect`? The reason I suggest that is because when consuming a hook, I'd expect it to re-render the consuming component every time the return value changes, which wouldn't be the case here.

When I added this hook it was used in both `AuthScreens` and `ReportScreen`, but it looks like the usage in `AuthScreens` was removed sometime last week

Reply:
Something like what @fabioh8010 suggested [here](https://github.com/Expensify/App/pull/44681/files), except removing the hook entirely in favor of just a `useEffect`

Reply:
@roryabraham of course I can use the approach from [this PR](https://github.com/Expensify/App/pull/44681/files), but I need to ask @fabioh8010 if he is okay with that, as I would need to copy some of his code to this PR.

Reply:
Sure @kosmydel please use everything you may need from my PR, I was exploring an improvement last week but didn't have time to go back to it.